creator-node/src/snapbackSM/snapbackSM.test.js

```diff
@@ -0,0 +1,228 @@
+/** Unit tests for SnapbackSM module */
+
+const assert = require('assert')
+const proxyquire = require('proxyquire')
+
+const DBManager = require('../dbManager.js')
+const { logger } = require('../logging.js')
+
+const {
+  SyncMode,
+  computeSyncModeForUserAndReplica
+} = require('./computeSyncModeForUserAndReplica.js')
+describe('Test computeSyncModeForUserAndReplica()', () => {
+  let primaryClock,
+    secondaryClock,
+    primaryFilesHash,
+    secondaryFilesHash,
+    primaryFilesHashMock
+
+  // Can be anything for test purposes
+  const wallet = 'wallet'
+
+  it('Throws if missing or invalid params', async () => {
+    primaryClock = 10
+    secondaryClock = 10
+    primaryFilesHash = undefined
+    secondaryFilesHash = undefined
+
+    try {
+      await computeSyncModeForUserAndReplica({
+        wallet,
+        primaryClock,
+        secondaryClock,
+        primaryFilesHash,
+        secondaryFilesHash
+      })
+    } catch (e) {
+      assert.strictEqual(
+        e.message,
+        '[computeSyncModeForUserAndReplica] Error: Missing or invalid params'
+      )
+    }
+  })
+
+  it('Returns SyncMode.None if all data is equal', async () => {
+    primaryClock = 10
+    secondaryClock = primaryClock
+    primaryFilesHash = '0x123'
+    secondaryFilesHash = primaryFilesHash
+
+    const syncMode = await computeSyncModeForUserAndReplica({
+      wallet,
+      primaryClock,
+      secondaryClock,
+      primaryFilesHash,
+      secondaryFilesHash
+    })
+
+    assert.strictEqual(syncMode, SyncMode.None)
+  })
+
+  it('Returns SyncMode.None if clocks and filesHashes equal', async () => {
+    primaryClock = 10
+    secondaryClock = primaryClock
+    primaryFilesHash = '0x123'
+    secondaryFilesHash = primaryFilesHash
+
+    const syncMode = await computeSyncModeForUserAndReplica({
+      wallet,
+      primaryClock,
+      secondaryClock,
+      primaryFilesHash,
+      secondaryFilesHash
+    })
+
+    assert.strictEqual(syncMode, SyncMode.None)
+  })
```

Comment:
aren't these the same tests? if so, could we just update the description for one of them to include the other?

Reply:
lolll great catch, idk how i did this. removed the duplicate!

thanks for actually reading through all the tests 🙂